mhs-reference-implementation/mhs/handler/tests/test_async_response_handler.py

```diff
@@ -37,10 +47,21 @@ def test_post(self, mock_get_uuid, mock_get_timestamp):
         mock_get_timestamp.return_value = "2012-03-15T06:51:08Z"
         expected_response = FileUtilities.get_file_string(str(self.message_dir / EXPECTED_RESPONSE_FILE))
         request_body = FileUtilities.get_file_string(str(self.message_dir / REQUEST_FILE))
+        mock_callback = Mock()
+        self.callbacks[REF_TO_MESSAGE_ID] = mock_callback
 
-        headers = {"Content-Type": 'multipart/related; boundary="--=_MIME-Boundary"'}
-        response = self.fetch("/", method="POST", body=request_body, headers=headers)
+        response = self.fetch("/", method="POST", body=request_body, headers=CONTENT_TYPE_HEADERS)
 
         self.assertEqual(response.code, 200)
         self.assertEqual(response.headers["Content-Type"], "text/xml")
         XmlUtilities.assert_xml_equal(expected_response, response.body)
+        mock_callback.assert_called_with(EXPECTED_MESSAGE)
```

Comment:
Do you think it's work asserting that whatever the `_send_ack` is doing was also called?

Reply:
`_send_ack` builds and sends the acknowledgement message in response to the request, so I think the check that `expected_response` equals `response.body` covers that, doesn't it?

Reply:
ah ok I see. So would the callback also generate another response? I can see you assert the callback is called. I think perhaps just making it clear that the expected_response and the response here are the acknowledgement responses and not the callback response.

Reply:
No, the callback doesn't return anything, which is why I'm only asserting that it's called. I'll update the name of the `expected_response` and `response` variables to (hopefully) make it clear that they are the acknowledgement responses.